components/script/dom/document.rs

```diff
@@ -2681,6 +2679,10 @@ impl DocumentMethods for Document {
         let window = window_from_node(self);
         let viewport = window.window_size().unwrap().visible_viewport;
 
+        if !self.browsing_context().is_some() {
```

Comment:
Is this correct? If it's correct, should I combine this with ` if x < 0.0 || y < 0.0 || x > viewport.width.get() || y > viewport.height.get() {`?